packages/assistify-ai/server/lib/SmartiAdapter.js

```diff
@@ -238,6 +243,33 @@ export class SmartiAdapter {
 		SystemLogger.debug('Smarti - retieved analysis result =', JSON.stringify(analysisResult, null, 2));
 		SystemLogger.debug(`Smarti - analysis complete -> update cache and notify room [ ${ roomId } ] for conversation [ ${ conversationId } ]`);
 		RocketChat.Notifications.notifyRoom(roomId, 'newConversationResult', analysisResult);
+
+		// update the affected message with the tokens extracted. This will trigger a re-rendering of the message
+		SmartiAdapter.updateTokensInMessages(roomId, analysisResult);
+	}
+
+	static updateTokensInMessages(roomId, analysisResult) {
+		const allTerms = analysisResult.tokens.reduce((terms, token) => terms.add(token.value), new Set());
```

Comment:
Couldn't we just use `analysisResult.tokens` here? It seems to be an array, and `forEach` which is used on `allTerms` in line 261 also works on those as well.

Reply:
I wanted to have a `Set` in order to ensure removed duplicates and check value with `has`